### ISSUE TITLE
Rust support

### ISSUE THREAD

**Issue by LucvandenBrand** [LucvandenBrand/WiiDevEnv#1]
As per request, I'm trying to see if this setup can support Rust. It would be nice if we would end up with:

- Gameplay programming in Rust.
   - It's fine if supporting code is in C++
   - Bevy seems like a great library to integrate. It's modular so we should be able to pick just what we need (and skip things like graphics, input and audio). 
- Nothing to install or do: automatic setup and build using Docker.
- Simple unassuming setup, free for use during a game jam.

**My hunches as to get a proper solution** can be found in these two articles:
1. https://docs.rust-embedded.org/book/interoperability/rust-with-c.html
2. https://docs.rust-embedded.org/book/interoperability/c-with-rust.html

If we can build a supporting set of precompiled C++ libraries and create some mapping methods to Rust, we can use them in our Rust gameplay code. Main could even be implemented in C++, simply kick-starting a main loop that calls our Rust code.  Our Rust code will then be compiled with the precompiled C++ code, which will give us the final Wii compatible binaries. **This will be transparent to the developer, who will happily code a game in Rust without the need to do much more.**

In other words, I think the proper solution lies in getting all Hardware specific code done in C++ (it's much more mature for it, much less a pain), and then to isolate the gameplay code and Write it in Rust. Almost like Rust will be our "scripting" language.

**Comment by ProfElements** [comment]
This may be out of nowhere but there are currently projects for rust on the wii already

[ogc-rs](https://github.com/rust-wii/ogc-rs)
A set of bindings / reimplementation of libogc for the wii

[rosalina](https://github.com/profelements/rosalina)
My project trying to make my own set of rusty wii stuff 

Again sorry for coming out of nowhere

**Comment by LucvandenBrand** [comment]
Thanks @ProfElements ! [ogc-rs](https://github.com/rust-wii/ogc-rs) has been really useful!
